File: project-manager/src/main/java/uni/lodz/pl/projectmanager/sprint/model/Sprint.java
package uni.lodz.pl.projectmanager.sprint.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import uni.lodz.pl.projectmanager.project.model.Project;

import javax.persistence.*;
import java.time.LocalDate;

@Getter
@Setter
@Entity
@NoArgsConstructor
public class Sprint {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Long id;
    @Column(nullable = false)
    private String name;
    @Column(nullable = false)
    private LocalDate start;
    @Column(nullable = false)
    private LocalDate end;
    @ManyToOne(optional = false)
    private Project project;
    @Column(nullable = false)
    private boolean closed;

    public Sprint(String name, LocalDate start, LocalDate end, Project project) {
        this.name = name;
        this.start = start;
        this.end = end;
        this.project = project;
        this.closed = false;
    }
}

File: project-manager-ui/src/model/Sprint.ts
import {Project} from "./Project";
import {SprintDto} from "./SprintDto";

export class Sprint {
  id: number;
  name: string;
  start: Date | null;
  end: Date | null;
  project: Project;

  constructor(sprintDto: SprintDto) {
    this.id = sprintDto.id;
    this.name = sprintDto.name;
    const s = sprintDto.start;
    this.start = s !== null ? new Date(s[0], s[1], s[2]) : null;
    const e = sprintDto.end;
    this.end = e !== null ? new Date(e[0], e[1], e[2]) : null;
    this.project = sprintDto.project;
    return;
  }
}